repo_policy_compliance/check.py

```diff
@@ -97,10 +101,20 @@ def wrapper(*args: P.args, **kwargs: P.kwargs) -> R | Report:
         """
         try:
             return func(*args, **kwargs)
-        except GithubClientError as exc:
-            return Report(result=Result.ERROR, reason=str(exc))
-        except ConfigurationError as exc:
+        except RetryableGithubClientError as exc:
             return Report(result=Result.ERROR, reason=str(exc))
```

Comment:
Switching to the end user perspective, I also feel that "The github client is returning an Rate Limit Exceeded error please wait before retrying" could be improved to something like "Checking repository compliance policy failed due to rate limit exceeded. Please wait before retrying".